testutil/ssntp_test_certs.go

```diff
@@ -486,3 +495,123 @@ func RoleToTestCert(role ssntp.Role) string {
 
 	return TestCertUnknown
 }
+
+// RoleToTestCertPath returns a string containing the path to a test
+// ca cert file, a string containing the path to the cert file for
+// the given role
+func RoleToTestCertPath(role ssntp.Role) (string, string, error) {
+	certdir, err := os.Getwd()
+	if err != nil {
+		return "", "", errors.New("Unable to get current directory")
+	}
+	cacert := path.Join(certdir, "CAcert-localhost.pem")
+	roleCert := ""
+
+	switch role {
+	case ssntp.SCHEDULER:
+		roleCert = path.Join(certdir, "cert-Scheduler-localhost.pem")
+	case ssntp.AGENT:
+		roleCert = path.Join(certdir, "cert-CNAgent-localhost.pem")
+	case ssntp.Controller:
+		roleCert = path.Join(certdir, "cert-Controller-localhost.pem")
+	case ssntp.CNCIAGENT:
+		roleCert = path.Join(certdir, "cert-CNCIAgent-localhost.pem")
+	case ssntp.NETAGENT:
+		roleCert = path.Join(certdir, "cert-NetworkingAgent-localhost.pem")
+	}
+
+	if roleCert == "" {
+		err = errors.New("No cert for role")
+	}
+
+	return cacert, roleCert, err
+}
+
+func makeTestCert(role ssntp.Role) error {
+	caPath, path, err := RoleToTestCertPath(role)
+	if err != nil {
+		return fmt.Errorf("Missing cert path: %v", err)
+	}
+
+	template, err := certs.CreateCertTemplate(role, "test", "test@test.test", []string{"localhost"}, []string{"127.0.0.1"})
+	if err != nil {
+		return fmt.Errorf("Unable to create cert template: %v", err)
+	}
+
+	writer, err := os.Create(path)
+	if err != nil {
+		return fmt.Errorf("Unable to create cert file: %v", err)
+	}
+	defer writer.Close()
+	if role == ssntp.SCHEDULER {
+		caWriter, err := os.Create(caPath)
+		if err != nil {
+			return fmt.Errorf("Unable to create ca file: %v", err)
+		}
+		defer caWriter.Close()
+		err = certs.CreateAnchorCert(template, true, writer, caWriter)
+		if err != nil {
+			return fmt.Errorf("Unable to populate cert file: %v", err)
+		}
+	} else {
+		_, schedulerPath, err := RoleToTestCertPath(ssntp.SCHEDULER)
+		if err != nil {
+			return fmt.Errorf("Unable to get scheduler cert path: %v", err)
+		}
+		anchorCert, err := ioutil.ReadFile(schedulerPath)
+		if err != nil {
+			return fmt.Errorf("Unable to read scheduler cert file: %v", err)
+		}
+		err = certs.CreateCert(template, true, anchorCert, writer)
+		if err != nil {
+			return fmt.Errorf("Unable to populate cert file: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// MakeTestCerts will create test certificate files for all roles
+// that return valid paths from RoleToTestCertPath
+func MakeTestCerts() error {
+	err := makeTestCert(ssntp.SCHEDULER)
```

Comment:
I think it would be nicer if MakeTestCerts called RemoveTestCerts if it encounters an error.  This way clients don't need to do any clean up if the function fails which is probably what they would expect.  Currently, the example presented in the commit message leaks certs if there is an error.